packages/shared/src/BootcAPI.ts

```diff
@@ -19,12 +19,15 @@
 import type { BootcBuildInfo, BuildType } from './models/bootc';
 import type { ImageInfo, ImageInspectInfo, ManifestInspectInfo, ContainerInfo } from '@podman-desktop/api';
 import type { ExamplesList } from './models/examples';
+import type { CreateVmOptions, VmDetails } from '@crc-org/macadam.js';
 
 export abstract class BootcApi {
   static readonly CHANNEL: string = 'BootcApi';
   abstract checkPrereqs(): Promise<string | undefined>;
   abstract checkVMLaunchPrereqs(buildId: string): Promise<string | undefined>;
   abstract launchVM(buildId: string): Promise<void>;
+  abstract macadamCreateVM(options: CreateVmOptions): Promise<void>;
```

Comment:
Even though this is macadam API and impossible to hide the implementation completely, we don't prefix PD function like `pdPullImage`. Would prefer to keep this like createVM / listVMs.